code/modules/mob/mob_grab.dm

```diff
@@ -97,7 +97,7 @@
 						return */
 		X.visible_message("<span class='danger'>[X] starts to devour [pulled]!</span>", \
 		"<span class='danger'>You start to devour [pulled]!</span>", null, 5)
-		if(do_after(X, 50, FALSE, 5, BUSY_ICON_HOSTILE))
+		if(do_after(X, min(30, 200 - pulled.traumatic_shock), FALSE, 5, BUSY_ICON_HOSTILE)) //Devour delay scales inversely with victim's pain; 20 seconds max, 3 seconds min
```

Comment:
I think you want here a `max(30, 200- pulled.traumatic_shock)`, no? `CLAMP` would be safer.